src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp

```diff
@@ -734,54 +738,53 @@ BrgemmEmitter::BrgemmEmitter(dnnl::impl::cpu::x64::jit_generator* h, dnnl::impl:
-    K = A_shape[A_layout[3]];
-    M_blk = matmulOptimalM;
-    M_tail = M % M_blk;
+    m_M = C_shape[C_layout[2]];
+    m_K = A_shape[A_layout[3]];
+    m_M_blk = matmulOptimalM;
+    m_M_tail = m_M % m_M_blk;
     // B_shape[B_layout[3]]
-    N = C_shape[C_layout[3]];
+    m_N = C_shape[C_layout[3]];
 
     auto brg0Prc = InferenceEngine::details::convertPrecision(brgemm_node->get_input_element_type(0));
     auto brg1Prc = InferenceEngine::details::convertPrecision(brgemm_node->get_input_element_type(1));
     io_data_size = {brg0Prc.size(), brg1Prc.size(), brgemm_node->get_output_element_type(0).size()};
-    brg0VnniFactor = 4 / brg0Prc.size();
-    bool brgWithAMX = isAMXSupported && brg0Prc != Precision::FP32 && (K % brg0VnniFactor == 0) && (N % brg0VnniFactor == 0);
+    m_brg0VnniFactor = 4 / brg0Prc.size();
+    bool brgWithAMX = brgemm_node->is_amx();
 
-    with_scratch = brgemm_node->get_input_size() == 3;
-    with_comp = !brgWithAMX && brg0Prc == Precision::I8;
+    m_with_comp = brgemm_node->is_with_compensations();
+    m_with_scratch = brgemm_node->is_with_scratchpad();
+    OPENVINO_ASSERT((m_with_scratch && brgemm_node->get_input_size() == 3) || !m_with_scratch, "Brgemm with scratchpad expect 3 inputs");
 
-    N_blk = brg1Prc == Precision::FP32 ? N :
-            brg1Prc == Precision::BF16 ? 32 : 64;
-    N_tail = N % N_blk;
-    K_blk = brgWithAMX ? brg0Prc == Precision::BF16 ? 32 : 64
-                       : K;
-    K_tail = K % K_blk;
+    m_N_blk = brg1Prc == Precision::FP32 ? m_N :
+              brg1Prc == Precision::BF16 ? 32 : 64;
+    m_N_tail = m_N % m_N_blk;
+    m_K_blk = brgWithAMX ? brg0Prc == Precision::BF16 ? 32 : 64
+                         : m_K;
+    m_K_tail = m_K % m_K_blk;
 
     size_t brg0BaseIdx = -1;
     for (size_t m = 0; m < 2; m++) {
         for (size_t k = 0; k < 2; k++) {
             for (size_t n = 0; n < 2; n++) {
-                auto& brgemmCtx = brgCtxs0[getBrgIdx(m, k, n)];
+                auto& brgemmCtx = m_brgCtxs0[getBrgIdx(m, k, n)];
 
-                auto M_ = m ? M_tail
-                            : M < M_blk ? 0 : M_blk;
-                auto N_ = n ? N_tail : N - N_tail;
-                auto K_ = k ? K_tail : K - K_tail;
-                auto beta = k && brgCtxs0[getBrgIdx(m, 0, n)].K != 0 ? 1.0f : 0.0f;
+                auto M_ = m ? m_M_tail
+                            : m_M < m_M_blk ? 0 : m_M_blk;
+                auto N_ = n ? m_N_tail : m_N - m_N_tail;
+                auto K_ = k ? m_K_tail : m_K - m_K_tail;
+                auto beta = k && m_brgCtxs0[getBrgIdx(m, 0, n)].K != 0 ? 1.0f : 0.0f;
 
                 brgemmCtx.M = M_;
                 brgemmCtx.N = N_;
                 brgemmCtx.K = K_;
                 brgemmCtx.LDA = leading_dimensions[0];
-                brgemmCtx.LDB = brg1Prc == Precision::FP32 ? leading_dimensions[1] : rnd_up(N, N_blk);
+                brgemmCtx.LDB = brg1Prc == Precision::FP32 ? leading_dimensions[1] : rnd_up(m_N, m_N_blk);
```

Comment:
Just curious, why do LBD values depend on precision? Is it a feature of non fp-32 primitives?

Reply:
I transferred it from MHA custom implementation. I think it depends on precision because if there is FP32, we don't need `BrgemmCopyB` otherwise we need it. I think that we should here another condition. For example, `is_with_data_repacking()`. Thanks a lot for the catch!